accord-core/src/test/java/accord/impl/basic/SimulatedDelayedExecutorService.java

```diff
@@ -26,12 +31,105 @@
+    {
+        private final long sequenceNumber;
+        private final long periodMillis;
+        private long timeMillis;
+        private ScheduledTask(long sequenceNumber, long value, TimeUnit unit, Callable<T> fn)
+        {
+            super(fn);
+            this.sequenceNumber = sequenceNumber;
+            periodMillis = unit.toMillis(value);
+        }
+
+        @Override
+        public long getDelay(TimeUnit unit)
+        {
+            return unit.convert(timeMillis - pending.nowInMillis(), TimeUnit.MILLISECONDS);
+        }
+
+        @Override
+        public int compareTo(Delayed other)
+        {
+            if (other == this) // compare zero if same object
+                return 0;
+            if (other instanceof ScheduledTask) {
+                ScheduledTask<?> x = (ScheduledTask<?>)other;
+                long diff = timeMillis - x.timeMillis;
+                if (diff < 0)
+                    return -1;
+                else if (diff > 0)
+                    return 1;
+                else if (sequenceNumber < x.sequenceNumber)
+                    return -1;
+                else
+                    return 1;
+            }
+            long diff = getDelay(NANOSECONDS) - other.getDelay(NANOSECONDS);
+            return (diff < 0) ? -1 : (diff > 0) ? 1 : 0;
+        }
+
+        @Override
+        public void run()
+        {
+            boolean periodic = periodMillis != 0;
+            if (!periodic)
+            {
+                super.run();
+            }
+            else
+            {
+                // run without setting the result
+                try
+                {
+                    fn.call();
+                    long nowMillis = pending.nowInMillis();
+                    if (periodMillis < 0)
+                    {
+                        // scheduleWithFixedDelay
+                        timeMillis = nowMillis + (-periodMillis);
+                    }
+                    else
+                    {
+                        // scheduleAtFixedRate
+                        timeMillis = timeMillis + periodMillis;
+                    }
+                    long delayMillis = timeMillis - nowMillis;
+                    if (delayMillis < 0)
+                        delayMillis = 0;
+                    schedule(this, delayMillis, TimeUnit.MILLISECONDS);
+                }
+                catch (Throwable t)
+                {
+                    // setSuccess may trigger callbacks, which may throw exceptions... if that happens then the result was already set
+                    if (!tryFailure(t))
+                    {
+                        throw new RuntimeException(t);
+                    }
+                }
+            }
+        }
+
+        @Override
+        public boolean cancel(boolean mayInterruptIfRunning)
+        {
+            boolean c = super.cancel(mayInterruptIfRunning);
+            if (c)
+                pending.remove(this);
```

Comment:
Removing it from the queue is conditional, `super.cancel` happens to be hard coded to return `false`, so cancel won't really work and also I don't see cancellation being persisted so a task will reschedule itself indefinitely after `cancel` is called.

Maybe it should check cancellation before starting execution as well as after execution completes.

Reply:
> super.cancel happens to be hard coded to return false

since this isn't generic and we know the subtype, switched to avoid calling `super.cancel`.  I changed it so the queue defines the return value; if we were able to remove it, then we return `true`, else `false`

> Maybe it should check cancellation before starting execution as well as after execution completes.

added